server/service/jct/api.coffee

```diff
@@ -736,38 +736,141 @@ API.service.jct.permission = (issn, institution) ->
+        # version - check for acceptable version
+        checks.version = if 'postprint' in permission.versions or 'publisher pdf' in permission.versions or 'acceptedVersion' in permission.versions or 'publishedVersion' in permission.versions then true else false
+        # requirements - check if there is no requirement or requirement matches Plan S
+        if permission.requirements?
+          if permission.requirements.funder? and permission.requirements.funder.length and 'Plan S' in permission.requirements.funder
+            checks.requirements = true
+            checks.planS = true
+          else
+            checks.requirements = false
+            checks.planS = false
+        else
+          checks.requirements = true
+          checks.planS = false
+        # check if embargo is 0 (if integer value)
+        if permission.embargo_months?
+          if typeof permission.embargo_months isnt 'number' or permission.embargo_months is 0
+            checks.embargo = true
           else
-            res.log.push code: 'SA.OABNonCompliant', parameters: licence: pbls, embargo: (if pb.embargo_months? then [pb.embargo_months] else undefined), version: pb.versions
+            checks.embargo = false
         else
-          res.log.push code: 'SA.OABNonCompliant', parameters: licence: pbls, embargo: (if pb.embargo_months? then [pb.embargo_months] else undefined), version: pb.versions
+          checks.embargo = true
+        # matched_license - get first matching license
+        if permission.licences? and permission.licences.length
+          for l in permission.licences ? []
+            if checks.matched_license is undefined and l.type.toLowerCase().replace(/\-/g,'').replace(/ /g,'') in ['ccby','ccbysa','cc0','ccbynd']
+              checks.matched_license = l.type
+        # license - check for matching license or missing license
+        if checks.matched_license or not permission.licences? or permission.licences.length is 0
+          checks.license = true
+        return checks
+
+      # get best permission based on score
+      _best_permission = (list_of_values) =>
+        if not list_of_values or list_of_values.length is 0
+          return undefined
+        best_score = 0
+        selected_value = undefined
+        for value in list_of_values
+          if value.score? and typeof value.score is 'number' and value.score > best_score
+            best_score = value.score
+            selected_value = value
+        if not selected_value
+          selected_value = list_of_values[0]
+        return selected_value
+
+      # evaluate each permission from OA works
+      oa_check =
+        OABCompliant: [],
+        OABIncomplete: [],
+        OABNonCompliant: []
+      for permission in perms.all_permissions
+        p_values = _gather_values(permission)
+        p_checks = _gather_checks(permission)
+        if p_checks.archive and \
+          p_checks.requirements and \
```

Comment:
This `if` statement doesn't take into account `p_checks.planS`, and I think it needs to.

The check should pass if:

1. `p_checks.requirements` is false
2. `p_checks.requirements` is true AND `p_checks.planS` is true

Reply:
No, that logic would be wrong. The plans check was superfluous anyway. I added that incase you wanted a qualification added. I then added that to the values extracted, as that was a better place. So I've just removed checks.plans so as to not confuse you